### PR TITLE
[New Feature] Add gradient accumulation and clip to support large batch size and stable training.

### PR DESCRIPTION
**Pull request by SamitHuang**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

Model like SVTR requires large batch-size training, which can be achieved by Gradient Accumulation

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

**Review by hqkate — APPROVED** [review]

File: mindocr/utils/train_step_wrapper.py [hunk 1]
```diff
@@ -100,16 +121,17 @@ def construct(self, *inputs):
...
 
-        # down-scale gradidents with loss_scale value. TODO: divide scaling_sense by accumulation steps for grad accumulate
-        grads = self.hyper_map(F.partial(_grad_scale, scaling_sens), grads)
+        # 2. down-scale gradients by loss_scale. grads = grads / scaling_sense  / grad_accu_steps
+        # also divide gradients by accumulation steps to avoid taking mean of  the accumulated gradients later
+        grads = self.hyper_map(F.partial(_grad_scale, scaling_sens * self.grad_accu_steps), grads)
 
-        # gradient reduction on distributed GPUs/NPUs
+        # 3. gradient reduction on distributed GPUs/NPUs
         grads = self.grad_reducer(grads)
```
Comment (zhtmike) [inline mindocr/utils/train_step_wrapper.py:132]: grad reducer 应该是只在参数更新时使用，否则batch gradient会和原本不一致

File: mindocr/utils/train_step_wrapper.py [hunk 2]
```diff
@@ -119,16 +141,29 @@ def construct(self, *inputs):
...
+                if self.cur_accu_step % self.grad_accu_steps == 0:
+                    # 6. clip grad
+                    if self.clip_grad:
+                        grads = ops.clip_by_global_norm(self.accumulated_grads, self.clip_norm)
+                    # 7. optimize
+                    loss = F.depend(loss, self.optimizer(grads))
+
+                    # clear gradient accumulation states
+                    loss = F.depend(loss, self.map(self.partial(ops.assign), self.accumulated_grads, self.zeros)) # self.accumulated_grads = 0
+                    loss = F.depend(loss, ops.assign(self.cur_accu_step, self.zero))      # self.cur_accu_step = 0
```
Comment (zhtmike) [inline mindocr/utils/train_step_wrapper.py:160]: lr rate 可能也需要更新。否则lr_scheduler steps和training steps 会不一致。
